viewer/db.js

```diff
@@ -1636,17 +1648,21 @@ Db.deleteFile = function (node, id, path, cb) {
 };
 
 Db.session2Sid = function (item) {
-  const ver = item._index.includes('sessions2') ? '2@' : '3@';
-  if (item._id.length < 31) {
+  // ver can be 2@ (sessions2), 3@ (sessions3), or x@ (user-specified queryExtraIndices)
+  const ver = internals.queryExtraIndicesRegex.some(re => re.test(item._index)) ? 'x@' : item._index.includes('sessions2') ? '2@' : '3@';
+  if (ver === 'x@') {
+    // document from queryExtraIndices, format Sid as x@_index:_id
+    return ver + item._index + ':' + item._id;
+  } else if (item._id.length < 31) {
```

Comment:
`Db.session2Sid` takes a result from a sessions search result set and formats an `Sid` string that indicates the version, index, and ID information. For documents from `queryExtraIndices` indices, format that string specifically as `x@`. Now, `ver` can be `2@` (sessions2), `3@` (sessions3), or `x@` (user-specified `queryExtraIndices`)